tests/ducktape/test_producer.py

```diff
@@ -0,0 +1,216 @@
+        self.logger.info("Verifying connection to external Kafka at localhost:9092")
+        
+        if not self.kafka.verify_connection():
+            raise Exception("Cannot connect to Kafka at localhost:9092. "
+                          "Please ensure Kafka is running.")
+        
+        self.logger.info("Successfully connected to Kafka")
+        
+    def test_basic_produce(self):
+        """Test basic message production to a topic"""
+        if Producer is None:
+            self.logger.error("confluent_kafka not available, skipping test")
+            return
+            
+        topic_name = "test-topic"
+        num_messages = 10
+        
+        # Create topic
+        self.kafka.create_topic(topic_name, partitions=1, replication_factor=1)
+        
+        # Wait for topic to be created
+        time.sleep(2)
+        
+        # Verify topic exists
+        topics = self.kafka.list_topics()
+        assert topic_name in topics, f"Topic {topic_name} was not created. Available topics: {topics}"
+        
+        # Configure producer
+        producer_config = {
+            'bootstrap.servers': self.kafka.bootstrap_servers(),
+            'client.id': 'ducktape-test-producer'
+        }
+        
+        self.logger.info("Creating producer with config: %s", producer_config)
+        producer = Producer(producer_config)
+        
+        # Track delivery results
+        delivered_messages = []
+        failed_messages = []
+        
+        def delivery_callback(err, msg):
+            """Delivery report callback"""
+            if err is not None:
+                self.logger.error("Message delivery failed: %s", err)
+                failed_messages.append(err)
+            else:
+                self.logger.debug("Message delivered to %s [%d] at offset %d",
+                                msg.topic(), msg.partition(), msg.offset())
+                delivered_messages.append(msg)
+        
+        # Produce messages
+        self.logger.info("Producing %d messages to topic %s", num_messages, topic_name)
+        for i in range(num_messages):
+            message_value = f"Test message {i}"
+            message_key = f"key-{i}"
+            
+            producer.produce(
+                topic=topic_name,
+                value=message_value,
+                key=message_key,
+                callback=delivery_callback
+            )
+            
+            # Poll to trigger delivery callbacks
+            producer.poll(0)
+            
+        # Flush to ensure all messages are sent
+        self.logger.info("Flushing producer...")
+        producer.flush(timeout=30)
+        
+        # Verify all messages were delivered
+        self.logger.info("Delivered: %d, Failed: %d", len(delivered_messages), len(failed_messages))
+        
+        assert len(failed_messages) == 0, f"Some messages failed to deliver: {failed_messages}"
+        assert len(delivered_messages) == num_messages, \
```

Comment:
TODO we'll want to extract execution metrics and assert against bounds therein. A follow-up effort